src/collections/Menu/Menu.js

```diff
@@ -145,28 +145,37 @@ class Menu extends Component {
   static Item = MenuItem
   static Menu = MenuMenu
 
-  renderItems() {
-    const { items, onItemClick } = this.props
+  componentWillMount() {
+    super.componentWillMount()
 
-    this.trySetState({ activeIndex: _.findIndex(items, ['active', false]) || -1 })
+    const { items } = this.props
+    if (items) this.trySetState({ activeIndex: _.findIndex(items, ['active', false]) || -1 })
+  }
 
-    return _.map(items, (index, item) => {
-      const { content, childKey, name, onClick, itemProps } = item
+  handleItemClick(e, { name, index }) {
+    const { onItemClick } = this.props
 
-      const active = this.state.activeIndex === index
-      const finalKey = childKey || [content, name].join('-')
+    this.trySetState({ activeIndex: index })
+    if (onItemClick) onItemClick(e, { name, index })
+  }
 
-      const handleClick = (e, { itemName, itemIndex }) => {
-        const handler = onItemClick || onClick
-        this.trySetState({ activeIndex: itemIndex })
+  renderItems() {
+    const { items } = this.props
+    const { activeIndex } = this.state
 
-        if (handler) handler(e, { itemName, itemIndex })
+    return _.map(items, (item, index) => {
+      const { content, childKey, name, onClick, itemProps } = item
+      const finalKey = childKey || [content, name].join('-')
+
+      const handleClick = (e, data) => {
+        this.handleItemClick(e, data)
+        if (onClick) onClick(e, data)
       }
```

Comment:
Sorry slow brain today, you are correct.  It looks like if you are going to call two functions, then we'll always have to create the callback function in every loop.  If we weren't calling both handleItemClick and onClick, then we could just pass `handleItemClick`.

As I write this, one other idea comes to mind.  You could just pass `onClick={this.handleItemClick}` which will get called with the item name/index.  Then, you could lookup the item in `items` and call its `onClick` with the same args if it exists:

```js
handleItemClick = (e, { name, index }) => {
  const { items, onItemClick } = this.props
  if (_.get(items[index], 'onClick')) items[index].onClick(e, { name, index })
  // ...
}
```